src/main/scala/fi/oph/koski/validation/JotpaValidation.scala

```diff
@@ -18,8 +18,41 @@ object JotpaValidation {
       oo.tila.opiskeluoikeusjaksot.map(validateOpiskeluoikeusjaksonRahoitusmuoto)
       :+ validateJaksojenRahoituksenYhtenäisyys(oo.tila.opiskeluoikeusjaksot)
       :+ validateJotpaRahoituksenAlkamispäivä(oo, jotpaRahoituksenRajapäivä)
+      :+ HttpStatus.fold(validateLaajuudet(oo))
     )
 
+  private def validateLaajuudet(oo: KoskeenTallennettavaOpiskeluoikeus): List[HttpStatus] = oo.suoritukset.map{
+    case s: VapaanSivistystyönJotpaKoulutuksenSuoritus if s.vahvistettu => validateSuorituksenLaajuus(s)
+    case _ => HttpStatus.ok
+  }
+
+  private def validateSuorituksenLaajuus(s: VapaanSivistystyönJotpaKoulutuksenSuoritus): HttpStatus = {
+    def validateOsasuorituksenLaajuus(
+      os: VapaanSivistystyönJotpaKoulutuksenOsasuorituksenSuoritus,
+      onAliOsasuoritus: Boolean = false
+    ): HttpStatus = {
+      HttpStatus.fold(
+        HttpStatus.validate(os.koulutusmoduuli.getLaajuus.isDefined)(
+          if(onAliOsasuoritus) {
+            KoskiErrorCategory.badRequest.validation.laajuudet(s"Vahvistetulta jatkuvaan oppimiseen suunnatulta vapaan sivistystyön koulutuksen aliosasuoritukselta puuttuu laajuus")
+          } else {
+            KoskiErrorCategory.badRequest.validation.laajuudet("Vahvistetulta jatkuvaan oppimiseen suunnatulta vapaan sivistystyön koulutuksen osasuoritukselta puuttuu laajuus")
+          }
+        ),
+        HttpStatus.fold(
+          os.osasuoritukset.getOrElse(List.empty).map(os => validateOsasuorituksenLaajuus(os, true))
```

Comment:
Suorituksella on metodi `osasuoritusLista`, joka tekee tämän listaksi muunnoksen.

Jos lisää tuon suorituksenTunniste:en, niin sitten voi ehkä harkita, että ei tarvitsisi edes tunnistaa aliosasuoritusta erikseen ja käyttää vaan `rekursiivisetOsasuoritukset`-metodia.

Reply:
En ollutkaan hoksannut näitä apufunktioita! Refaktoroin toteutuksen käyttämään rekursiivisetOsasuoritukset-metodia.